kani-compiler/src/kani_middle/transform/check_uninit/uninit_visitor.rs

```diff
@@ -0,0 +1,783 @@
+                                        operand: args[0].clone(),
+                                        count: args[2].clone(),
+                                        position: InsertPosition::Before,
+                                    });
+                                    self.push_target(SourceOp::Set {
+                                        operand: args[1].clone(),
+                                        count: args[2].clone(),
+                                        value: true,
+                                        position: InsertPosition::After,
+                                    });
+                                }
+                                "copy_nonoverlapping" => unreachable!(
+                                    "Expected `core::intrinsics::unreachable` to be handled by `StatementKind::CopyNonOverlapping`"
+                                ),
+                                "copysignf32"
+                                | "copysignf64"
+                                | "cosf32"
+                                | "cosf64"
+                                | "ctlz"
+                                | "ctlz_nonzero"
+                                | "ctpop"
+                                | "cttz"
+                                | "cttz_nonzero"
+                                | "discriminant_value"
+                                | "exact_div"
+                                | "exp2f32"
+                                | "exp2f64"
+                                | "expf32"
+                                | "expf64"
+                                | "fabsf32"
+                                | "fabsf64"
+                                | "fadd_fast"
+                                | "fdiv_fast"
+                                | "floorf32"
+                                | "floorf64"
+                                | "fmaf32"
+                                | "fmaf64"
+                                | "fmul_fast"
+                                | "forget"
+                                | "fsub_fast"
+                                | "is_val_statically_known"
+                                | "likely"
+                                | "log10f32"
+                                | "log10f64"
+                                | "log2f32"
+                                | "log2f64"
+                                | "logf32"
+                                | "logf64"
+                                | "maxnumf32"
+                                | "maxnumf64"
+                                | "min_align_of"
+                                | "min_align_of_val"
+                                | "minnumf32"
+                                | "minnumf64"
+                                | "mul_with_overflow"
+                                | "nearbyintf32"
+                                | "nearbyintf64"
+                                | "needs_drop" => {}
```

Comment:
I really appreciate the thoroughness. May I suggest that you move the cases that we don't have to do any work to a validation function? Maybe you can change the default cases to be something like:
```rust
intrinsic_name if unsupported_intrinsic(intrinsic_name) => {
        // and unsupported operation instead
    }
_ => { /* no-op  */ }
```
Where `unsupported_intrinsic()` checks if the name does not match any of the names you have listed here and in other places. BTW, I'm not great at names, so feel free to rename the function